### PR TITLE
merge flo-contrib and flo

### PR DESCRIPTION
**Pull request by honnix**
* clean up pom.xml
* pass typesafe config to TerminationHookFactory
* add flo-styx with TerminationLogging
* upgrade libs
* get logback completely out of the picture

@NatashaL @bergman PTAL

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -80,51 +90,87 @@
...
+        <groupId>com.google.cloud</groupId>
+        <artifactId>google-cloud-bigquery</artifactId>
+        <version>1.29.0</version>
+      </dependency>
+
+      <!-- version resolution -->
+      <dependency>
+        <groupId>com.google.guava</groupId>
+        <artifactId>guava</artifactId>
+        <version>25.0-jre</version>
```
Comment (bergman) [inline pom.xml]: this sounds way too recent, i've had problems with everything after 23 (when they added the `-jre` and `-android`)

Comment (honnix) [reply pom.xml]: That's more or less what we have in Styx production. :P
What kind of problems do we have?

File: contrib/flo-styx/src/main/java/com/spotify/flo/contrib/styx/TerminationLogging.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    this.config = Objects.requireNonNull(config);
+  }
+
+  @Override
+  public void accept(Integer exitCode) {
+    final String content = String.format("{\"component_id\": \"%s\","
+                                         + "\"workflow_id\": \"%s\","
+                                         + "\"parameter\": \"%s\","
+                                         + "\"execution_id\": \"%s\","
+                                         + "\"event\": \"terminated\","
```
Comment (NatashaL) [inline contrib/flo-styx/src/main/java/com/spotify/flo/contrib/styx/TerminationLogging.java]: I would assume we want `exited` here, not sure if it makes any difference for Styx. With edgedocker-base we log `terminated` (given SIGTERM), `interrupted` (given SIGINT) and `exited` (for natural termination).

Comment (honnix) [reply contrib/flo-styx/src/main/java/com/spotify/flo/contrib/styx/TerminationLogging.java]: right, that sounds better.

File: contrib/flo-bigquery/src/main/java/com/spotify/flo/contrib/bigquery/BigQueryContext.java [hunk 3]
```diff
@@ -135,4 +136,20 @@ TableId publish(StagingTableId stagingTableId) {
 
     return tableId;
   }
+
+  private static void checkJobStatus(Job job, TableId tableId) {
+    if (job != null && job.getStatus().getError() == null) {
```
[the remark this answers is not in the document]
Comment (honnix) [reply contrib/flo-bigquery/src/main/java/com/spotify/flo/contrib/bigquery/BigQueryContext.java]: By reading the code, I think it is still possible no exception thrown but the job failed. This API designing and doc are so confusing.

File: pom.xml (hunk 1 above)
Comment (honnix) [reply pom.xml]: downgraded to 23.0

**Comment by NatashaL** [comment]
@honnix the error handling seems like it covers all cases. Both if there's an exception thrown and if the error is contained. Don't see what more we can do about it for now. 

I think this is good for now. Any more comments or 👍  ? cc @bergman

**Pull request merged**